From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Rose <japthorp@slack-corp.com>
Date: Wed, 20 Mar 2024 15:25:40 -0700
Subject: Revert "[power] Update TRACE_COUNTER in power monitor"

This reverts commit dfb50c50fd84fafe1227bc84d08ff84fc2f0df21.

This broke because we build with use_perfetto_client_library = false due
to Node.js incompatabilities. However, this isn't sustainable and we
should build with use_perfetto_client_library = true (like upstream
does), and when we do that, we should remove this patch.

diff --git a/base/power_monitor/speed_limit_observer_win.cc b/base/power_monitor/speed_limit_observer_win.cc
index 700be2b553b35f58abd9b4007e4821ed9d91e9a8..f7f7927e4dbe7045b9103752b77f0b53b1002a32 100644
--- a/base/power_monitor/speed_limit_observer_win.cc
+++ b/base/power_monitor/speed_limit_observer_win.cc
@@ -144,14 +144,14 @@ int SpeedLimitObserverWin::GetCurrentSpeedLimit() {
   TRACE_EVENT_CATEGORY_GROUP_ENABLED(kPowerTraceCategory,
                                      &trace_events_enabled);
   if (trace_events_enabled) {
-    TRACE_COUNTER(kPowerTraceCategory, "idleness", idleness_percent);
-    TRACE_COUNTER(kPowerTraceCategory, "throttling_level",
-                  static_cast<unsigned int>(throttling_level * 100));
+    TRACE_COUNTER1(kPowerTraceCategory, "idleness", idleness_percent);
+    TRACE_COUNTER1(kPowerTraceCategory, "throttling_level",
+                   static_cast<unsigned int>(throttling_level * 100));
 
 #if defined(ARCH_CPU_X86_FAMILY)
     double cpu_frequency = EstimateCpuFrequency();
-    TRACE_COUNTER(kPowerTraceCategory, "frequency_mhz",
-                  static_cast<unsigned int>(cpu_frequency / 1'000'000));
+    TRACE_COUNTER1(kPowerTraceCategory, "frequency_mhz",
+                   static_cast<unsigned int>(cpu_frequency / 1'000'000));
 #endif
   }
 
@@ -198,8 +198,8 @@ void SpeedLimitObserverWin::OnTimerTick() {
     callback_.Run(speed_limit_);
   }
 
-  TRACE_COUNTER(kPowerTraceCategory, "speed_limit",
-                static_cast<unsigned int>(speed_limit));
+  TRACE_COUNTER1(kPowerTraceCategory, "speed_limit",
+                 static_cast<unsigned int>(speed_limit));
 }
 
 float SpeedLimitObserverWin::EstimateThrottlingLevel() {
@@ -259,7 +259,7 @@ float SpeedLimitObserverWin::EstimateThrottlingLevel() {
       << " num_non_idle_cpus:" << num_non_idle_cpus;
   throttling_level = (load_fraction_total / num_cpus());
 
-  TRACE_COUNTER(kPowerTraceCategory, "num_active_cpus", num_active_cpus);
+  TRACE_COUNTER1(kPowerTraceCategory, "num_active_cpus", num_active_cpus);
 
   return throttling_level;
 }
